src/colvarcomp.cpp

```diff
@@ -291,6 +297,105 @@ colvar::cvc::~cvc()
 }
 
 
+void colvar::cvc::register_atom_group(cvm::atom_group *ag)
+{
+  atom_groups.push_back(ag);
+  add_child(ag);
+}
+
+
+
+void colvar::cvc::register_param(std::string const &param_name, void *param_ptr,
+                                 colvarvalue *param_grad_ptr)
+{
+  param_map[param_name] = param_ptr;
+  if (param_grad_ptr) {
+    param_grad_map[param_name] = param_grad_ptr;
+  }
+}
+
+
+int colvar::cvc::param_exists(std::string const &param_name)
+{
+  if (param_map.count(param_name) > 0) {
+    return COLVARS_OK;
+  }
+  return INPUT_ERROR;
+}
+
+
+void const *colvar::cvc::get_param(std::string const &param_name)
+{
+  if (param_map.count(param_name) > 0) {
+    return param_map[param_name];
+  }
+  cvm::error("Error: parameter \""+param_name+"\" not found.\n", INPUT_ERROR);
+  return NULL;
+}
+
+
+std::vector<std::string> colvar::cvc::get_param_names()
+{
+  std::vector<std::string> result;
+  for (std::map<std::string, void const *>::const_iterator elem =
+         param_map.begin(); elem != param_map.end(); elem++) {
+    result.push_back(elem->first);
+  }
+  return result;
+}
+
+
+colvarvalue const *colvar::cvc::get_param_grad(std::string const &param_name)
+{
+  if (param_grad_map.count(param_name) > 0) {
+    return param_grad_map[param_name];
+  }
+  cvm::error("Error: gradient of parameter \""+param_name+"\" not found.\n",
+             INPUT_ERROR);
+  return NULL;
+}
+
+
+std::vector<std::string> colvar::cvc::get_param_grad_names()
+{
+  std::vector<std::string> result;
+  for (std::map<std::string, colvarvalue const *>::const_iterator elem =
+         param_grad_map.begin(); elem != param_grad_map.end(); elem++) {
+    result.push_back(elem->first);
+  }
+  return result;
+}
+
+
+int colvar::cvc::set_param(std::string const &param_name,
+                           void const *new_value)
+{
+  if (param_map.count(param_name) > 0) {
+
```

Comment:
Why not just iterating over the key value of `param_map` and comparing?
```
for (auto it = param_map.begin(); it != param_map.end(); ++param_map) {
  if (param_name.compare(it->first)) {
    // Does the second field of the map point to the variables sup_coeff, sup_np, period and wrap_center?
    it->second = new_value;
  }
}
```

Reply:
Unfortunately not all the parameters have the same type, so the assignment operator cannot be the same for all values. So several exceptions would have to crawl their way into that loop regardless. The best design is in my opinion the one you used already with `std::map<std::string, std::function>` but this part of the code cannot be conditionally compiled yet.